pkg/mdlsub/fixtures.go

```diff
@@ -0,0 +1,192 @@
+	"github.com/go-resty/resty/v2"
+	"github.com/justtrackio/gosoline/pkg/cfg"
+	"github.com/justtrackio/gosoline/pkg/encoding/json"
+	"github.com/justtrackio/gosoline/pkg/fixtures"
+	"github.com/justtrackio/gosoline/pkg/log"
+	"github.com/justtrackio/gosoline/pkg/refl"
+)
+
+type FixtureSettings struct {
+	DataSetDbName string `cfg:"data_set_db_name"`
+	Host          string `cfg:"host"`
+	Path          string `cfg:"path"`
+}
+
+type FetchResult struct {
+	spec        *ModelSpecification
+	contentType string
+	data        []byte
+}
+
+var unmarhaller = map[string]func(data []byte, v interface{}) error{
+	"application/json": json.Unmarshal,
+}
+
+func FixtureSetFactory(transformerFactoryMap TransformerMapTypeVersionFactories) fixtures.FixtureSetsFactory {
+	return func(ctx context.Context, config cfg.Config, logger log.Logger) ([]fixtures.FixtureSet, error) {
+		var err error
+		var core SubscriberCore
+		var sets []fixtures.FixtureSet
+
+		settings := unmarshallSettings(config)
+
+		fixtureSettings := &FixtureSettings{}
+		config.UnmarshalKey("fixtures.loaders.api", fixtureSettings)
+
+		if core, err = NewSubscriberCore(ctx, config, logger, settings.Subscribers, transformerFactoryMap); err != nil {
+			return nil, fmt.Errorf("failed to create subscriber core: %w", err)
+		}
+
+		for _, sub := range settings.Subscribers {
+			sets = append(sets, NewFixtureSet(logger, sub.SourceModel, core, fixtureSettings))
+		}
+
+		return sets, nil
+	}
+}
+
+type FixtureSet struct {
+	logger     log.Logger
+	source     SubscriberModel
+	core       SubscriberCore
+	httpClient *resty.Client
+	settings   *FixtureSettings
+}
+
+func NewFixtureSet(logger log.Logger, source SubscriberModel, core SubscriberCore, settings *FixtureSettings) *FixtureSet {
+	return NewFixtureSetWithInterfaces(logger, source, core, settings, resty.New())
+}
+
+func NewFixtureSetWithInterfaces(logger log.Logger, source SubscriberModel, core SubscriberCore, settings *FixtureSettings, httpClient *resty.Client) *FixtureSet {
+	return &FixtureSet{
+		logger:     logger.WithChannel("fixtures"),
+		source:     source,
+		core:       core,
+		httpClient: httpClient,
+		settings:   settings,
+	}
+}
+
+func (f FixtureSet) Write(ctx context.Context) error {
+	var ok bool
+	var err error
+	var res *FetchResult
+	var transformer ModelTransformer
+	var unmarshal func(data []byte, v interface{}) error
+	var output Output
+	var items []interface{}
+	var model Model
+
+	if res, err = f.Fetch(ctx); err != nil {
+		return fmt.Errorf("failed to fetch model %s: %w", f.source.String(), err)
+	}
+
+	if transformer, err = f.core.GetTransformer(res.spec); err != nil {
+		return fmt.Errorf("failed to get transformer: %w", err)
+	}
+
+	input := transformer.GetInput()
+	slice := refl.CreatePointerToSliceOfTypeAndSize(input, 0)
+
+	if unmarshal, ok = unmarhaller[res.contentType]; !ok {
+		return fmt.Errorf("unknown content type: %s", res.contentType)
+	}
+
+	if err = unmarshal(res.data, slice); err != nil {
```

Comment:
As discussed, please don't directly unmarshal the response body. First unmarshal into a 
```go
type x struct{
    Data []any `json:"data"`
}
```

Reply:
adjusted my original comment to refer to json.RawMessage instead of []any